controller/sync.go

```diff
@@ -167,6 +171,8 @@ func (m *appStateManager) SyncAppState(app *v1alpha1.Application, state *v1alpha
 		return
 	}
 
+	atomic.AddUint64(&syncCounter, 1)
+	syncId := fmt.Sprintf("%05d-%s", syncCounter, rand.RandString(5))
```

Comment:
create a unique ID for the sync, note I looked into OpenTracing, but it was a lot of effort